neo/api/utils.py

```diff
@@ -1,24 +1,46 @@
 import json
+import gzip
 from functools import wraps
 
+COMPRESS_FASTEST = 1
+BASE_STRING_SIZE = 49
+MTU_TCP_PACKET_SIZE = 1500
+COMPRESS_THRESHOLD = MTU_TCP_PACKET_SIZE + BASE_STRING_SIZE
+
 
 # @json_response decorator for class methods
 def json_response(func):
     """ @json_response decorator adds header and dumps response object """
+
     @wraps(func)
     def wrapper(self, request, *args, **kwargs):
         res = func(self, request, *args, **kwargs)
+        response_data = json.dumps(res) if isinstance(res, dict) else res
         request.setHeader('Content-Type', 'application/json')
-        return json.dumps(res) if isinstance(res, dict) else res
+
+        if len(response_data) > COMPRESS_THRESHOLD:
+            accept_encoding = request.getHeader('Accept-Encoding')
+            if accept_encoding:
+                encodings = accept_encoding.split(',')
```

Comment:
i think this won't be enought, because the header might have spaces too: `Accept-Encoding: deflate, gzip;q=1.0, *;q=0.5`

Perhaps update to `encodings = [encoding.strip() for encoding in accept_encoding.split(',')`]

Reply:
Oh, shit this doesn't even fully work, because the gzip part may be ` gzip;q=1.0, *;q=0.5` (so a space and then more options)

Reply:
Not sure if we want to support the gzip options. If not, we might use this:

    use_gzip = any("gzip" in encoding for for encoding in accept_encoding.split(',')